tools/expotools/src/commands/CreateUnimodule.ts

```diff
@@ -1,21 +1,45 @@
-import path from 'path';
-import chalk from 'chalk';
-import JsonFile from '@expo/json-file';
 import { Command } from '@expo/commander';
+import JsonFile from '@expo/json-file';
 import spawnAsync from '@expo/spawn-async';
+import chalk from 'chalk';
+import path from 'path';
 
 import { PACKAGES_DIR, EXPO_DIR } from '../Constants';
 
 type ActionOptions = {
   name: string;
+  template?: string;
+  useLocalTemplate?: boolean;
 };
 
-async function generateModuleWithExpoCLI(unimoduleDirectory) {
+const TEMPLATE_PACKAGE_NAME = 'expo-module-template';
+
+async function generateModuleWithExpoCLI(
+  unimoduleDirectory: string,
+  options: Pick<ActionOptions, 'template' | 'useLocalTemplate'>
+) {
   console.log(
     `Creating new unimodule under ${chalk.magenta(path.relative(EXPO_DIR, unimoduleDirectory))}...`
   );
 
-  await spawnAsync('expo', ['generate-module', unimoduleDirectory], {
+  const { template, useLocalTemplate } = options;
+  const templateParams: string[] = [];
+
+  if (template) {
+    console.log(`Using custom module template: ${chalk.blue(template)}`);
+
+    templateParams.push('--template', template);
+  } else if (useLocalTemplate) {
+    const templatePath = path.join(PACKAGES_DIR, TEMPLATE_PACKAGE_NAME);
+
+    console.log(
+      `Using local module template from ${chalk.blue(path.relative(EXPO_DIR, templatePath))}`
+    );
+
+    templateParams.push('--template', templatePath);
+  }
+
+  await spawnAsync('expo', ['generate-module', ...templateParams, unimoduleDirectory], {
```

Comment:
You can also use `spawnAsync` from `Utils.ts` which defaults to `cwd: EXPO_DIR` and also passes all env variables 😉

Reply:
That's not a bad idea, especially it'd fit my plans of enhancing `expo-cli generate-module` command. If using this custom `spawnAsync` would allow to pass other params (not specified explicitly here) to the CLI command, that's worth implementing. However I'd still want to intercept the `--template` option here.